warehouse/query-core/src/main/java/datawave/query/tables/ShardQueryLogic.java

```diff
@@ -746,7 +746,32 @@ protected void loadQueryParameters(ShardQueryConfiguration config, Query setting
 
         if (StringUtils.isNotBlank(typeList)) {
             HashSet<String> typeFilter = new HashSet<>();
-            typeFilter.addAll(Arrays.asList(StringUtils.split(typeList, Constants.PARAM_VALUE_SEP)));
+            HashSet<String> excludeSet = new HashSet<>();
+
+            for (String dataType : Arrays.asList(StringUtils.split(typeList, Constants.PARAM_VALUE_SEP))) {
+                if (dataType.charAt(0) == '!') {
+                    excludeSet.add(StringUtils.substring(dataType, 1));
+                }
+                else {
+                    typeFilter.add(dataType);
+                }
+            }
+
+            if (!excludeSet.isEmpty()) {
+                if (typeFilter.isEmpty()) {
+                    MetadataHelper metadataHelper = prepareMetadataHelper(config.getClient(), this.getMetadataTableName(), config.getAuthorizations(), config.isRawTypes());
+
+                    try {
+                        for (Type<?> type : metadataHelper.getAllDatatypes()) {
```

Comment:
These are the wrong data types.  I know this is highly confusing, but these data types are the types assigned to fields (e.g. LcNoDiacritics).  Unfortunately we chose the initial "datatype" designation unwisely.  So "datatype" really needs a qualifier of some sort.  Field datatype vs event or ingest datatype ?  IngestDataTypes is used for various method parameters in the MetadataHelper.   Anyway, the method metadataHelper.getDatatypes((Set<String>)null) will get the data types you are looking for.